### PR TITLE
[Serve] [Docs] Update the "Monitoring Ray Serve" Page

### LINKED ISSUE #27720: [Serve] [Docs] Update metrics that Ray Serve offers in documentation
### Description

Ray Serve surfaces system metrics for debugging. [The metrics documentation](https://docs.ray.io/en/releases-1.13.0/serve/monitoring.html#metrics) is outdated and contains metrics that no longer exist.

Screen Shot 2022-08-09 at 3 31 02 PM

For example, the metrics `serve_backend_queuing_latency_ms` and `serve_replica_queued_queries` seem to have been removed in #15065. However, they're still documented.

### Link

[Link on the master branch](https://docs.ray.io/en/master/serve/monitoring.html#metrics)

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

The "Monitoring Ray Serve" page explains how to inspect your Ray Serve applications. This change updates the page to remove outdated metrics that Serve no longer exposes and to upgrade code samples to use 2.0 APIs. It also improves the content's readability and organization.

Link to updated "Monitoring Ray Serve" page: https://ray--27777.org.readthedocs.build/en/27777/serve/monitoring.html

## Related issue number

Closes #27720.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change adds assertions to the documentation code, so they're tested in CI.

### CODE REVIEW

File: doc/source/serve/monitoring.md [hunk 1]
```diff
@@ -128,17 +145,17 @@ clients:
...
+  - job_name: ray
+    static_configs:
+      - labels:
+        job: ray
+        __path__: /tmp/ray/session_latest/logs/serve/*.*
 ```
 
-The relevant part for Ray is the `static_configs` field, where we have indicated the location of our log files with `__path__`.
+The relevant part for Ray Serve is the `static_configs` field, where we have indicated the location of our log files with `__path__`.
 The expression `*.*` will match all files, but not directories, which cause an error with Promtail.
```
Comment (shrekris-anyscale) [inline doc/source/serve/monitoring.md:156]: What does this sentence mean? Is it saying that directories in general cause an error with Promtail? Or is it saying that the expression `*.*` matches all files (but not directories) that cause an error with Promtail?

Comment (edoakes) [reply doc/source/serve/monitoring.md:156]: Hmm I'm not actually sure but I'd interpret it as "this matches all files, but doesn't match any directories, which causes an error with Promtail."

I think it was written by @simon-mo or @architkulkarni ?

**Comment by edoakes** [comment]
@shrekris-anyscale FYI you have a merge conflict here

File: doc/source/serve/monitoring.md (hunk 1 above)
Comment (architkulkarni) [reply doc/source/serve/monitoring.md:156]: I wrote this a while ago, I believe it means that directories cause an error with promtail, so use `*.*` to not match directories

Comment (simon-mo) [reply doc/source/serve/monitoring.md:156]: yup @architkulkarni is correct

Comment (shrekris-anyscale) [reply doc/source/serve/monitoring.md]: Thanks, I revised the wording.

File: doc/source/serve/monitoring.md [hunk 2]
```diff
@@ -2,74 +2,107 @@
...
-A high-level way to monitor your Ray Serve application is via the Ray Dashboard.
-See the [Ray Dashboard documentation](ray-dashboard) for a detailed overview, including instructions on how to view the dashboard.
+## Ray dashboard
 
-Below is an example of what the "Actors" tab of the Ray Dashboard might look like for a running Serve application:
+You can use the Ray dashboard to get a high-level overview of your Ray cluster and Ray Serve application's states.
+This includes details such as:
+* the number of deployment replicas currently running
+* logs for your Serve controller, deployment replicas, and HTTP proxies
+* the Ray nodes (i.e. machines) running in your Ray cluster
```
Comment (edoakes) [inline doc/source/serve/monitoring.md:17]: period to end the last bullet point

File: doc/source/serve/monitoring.md [hunk 3]
```diff
@@ -2,74 +2,107 @@
...
+
+You can access the Ray dashboard at port 8265 at your cluster's URI.
+For example, if you're running Ray Serve on a local Ray cluster, you can access the dashboard by going to this address in your browser:
+
+```
+http://localhost:8265
+```
```
Comment (edoakes) [inline doc/source/serve/monitoring.md:24]:
```suggestion
For example, if you're running Ray Serve locally, you can access the dashboard by going to `http://localhost:8265` in your browser.
```

save some page real estate

File: doc/source/serve/monitoring.md [hunk 4]
```diff
@@ -2,74 +2,107 @@
...
+
+We can view important information about our application here.
+For example, we can inspect our deployment replicas by navigating to the Ray dashboard's "Actors" tab while our Serve application is running:
```
Comment (edoakes) [inline doc/source/serve/monitoring.md:27]: the previous paragraph used "you" and now you're using "we," stick with one

**Review by edoakes — COMMENTED** [review]
Thanks for adding the `serve.run` changes, those will be very helpful.

**Comment by shrekris-anyscale** [comment]
@edoakes Thanks for [the review](https://github.com/ray-project/ray/pull/27777#pullrequestreview-1070298557)! I addressed your comments.

**Review by edoakes — APPROVED** [review]

File: doc/source/serve/monitoring.md [hunk 5]
```diff
@@ -82,18 +111,21 @@ class Silenced:
         logger.setLevel(logging.ERROR)
 ```
 
-This will prevent the replica INFO-level logs from being written to STDOUT or to files on disk.
+This controls which logs are written to STDOUT or files on disk.
 You can also use your own custom logger, in which case you'll need to configure the behavior to write to STDOUT/STDERR, files on disk, or both.
```
Comment (dmatrix) [inline doc/source/serve/monitoring.md:116]: Apart from standard Python logger, Serve supports custom logging. Custom logging via configuration can control the behavior of what messages are written to STDOUT/STDERR, files on disk, or both.

File: doc/source/serve/monitoring.md [hunk 6]
```diff
@@ -82,18 +111,21 @@ class Silenced:
...
 
-Here is a quick walkthrough of how to explore and filter your logs using [Loki](https://grafana.com/oss/loki/).
-Setup and configuration is very easy on Kubernetes, but in this tutorial we'll just set things up manually.
+(serve-logging-loki)=
+### Filtering logs with Loki
 
-First, install Loki and Promtail using the instructions on <https://grafana.com>.
-It will be convenient to save the Loki and Promtail executables in the same directory, and to navigate to this directory in your terminal before beginning this walkthrough.
+You can explore and filter your logs using [Loki](https://grafana.com/oss/loki/).
+Setup and configuration is straightforward on Kubernetes, but as a tutorial, let's set up Loki manually.
```
Comment (dmatrix) [inline doc/source/serve/monitoring.md:124]: Is "setup and configuration" considered a singular compound subject (like ham and cheese is great for breakfast) in this context? If not perhaps change to

"Setup and configuration are straightforward on Kubernetes."

File: doc/source/serve/monitoring.md [hunk 7]
```diff
@@ -163,11 +198,14 @@ You should see something similar to the following:
 :align: center
```
Comment (dmatrix) [inline doc/source/serve/monitoring.md:198]: passward "admin."

File: doc/source/serve/monitoring.md [hunk 8]
```diff
@@ -213,16 +247,19 @@ ray start --head --metrics-export-port=8080
...
 
-```{literalinclude} ../../../python/ray/serve/examples/doc/snippet_metrics.py
+```{literalinclude} doc_code/monitoring/metrics_snippet.py
+:start-after: __start__
+:end-before: __end__
+:language: python
 ```
 
-The requests will loop and can be canceled with `Ctrl-C`. 
+The requests will loop and can be canceled with `ctrl-c`.
```
Comment (dmatrix) [inline doc/source/serve/monitoring.md:257]: The requests will loop until canceled with `ctrl-c`.

**Review by dmatrix — APPROVED** [review]
With those changes, it should be good to go.

Just a general comment. If possible for 2.1, we should consider providing dashboards for this; there is too much boiler code they have to write. But at least we are collecting stats; matter of accessing and presenting easily.

**Comment by shrekris-anyscale** [comment]
Thanks for the [review](https://github.com/ray-project/ray/pull/27777#pullrequestreview-1070394225) @dmatrix! I've addressed all your comments.

> Just a general comment. If possible for 2.1, we should consider providing dashboards for this; there is too much boiler code they have to write. But at least we are collecting stats; matter of accessing and presenting easily.

Yeah that's a good suggestion. I think that would improve Ray Serve's observability.

**Pull request merged**